api/data/notifications.go

```diff
@@ -0,0 +1,163 @@
+		Events            []string `xml:"Event" json:"Events"`
+		Filter            Filter   `xml:"Filter" json:"Filter"`
+	}
+
+	Filter struct {
+		Key Key `xml:"S3Key" json:"S3Key"`
+	}
+
+	Key struct {
+		FilterRules []FilterRule `xml:"FilterRule" json:"FilterRules"`
+	}
+
+	FilterRule struct {
+		Name  string `xml:"Name" json:"Name"`
+		Value string `xml:"Value" json:"Value"`
+	}
+)
+
+var validEvents = map[string]struct{}{
+	"s3:ReducedRedundancyLostObject":                   {},
+	"s3:ObjectCreated:*":                               {},
+	"s3:ObjectCreated:Put":                             {},
+	"s3:ObjectCreated:Post":                            {},
+	"s3:ObjectCreated:Copy":                            {},
+	"s3:ObjectCreated:CompleteMultipartUpload":         {},
+	"s3:ObjectRemoved:*":                               {},
+	"s3:ObjectRemoved:Delete":                          {},
+	"s3:ObjectRemoved:DeleteMarkerCreated":             {},
+	"s3:ObjectRestore:*":                               {},
+	"s3:ObjectRestore:Post":                            {},
+	"s3:ObjectRestore:Completed":                       {},
+	"s3:Replication:*":                                 {},
+	"s3:Replication:OperationFailedReplication":        {},
+	"s3:Replication:OperationNotTracked":               {},
+	"s3:Replication:OperationMissedThreshold":          {},
+	"s3:Replication:OperationReplicatedAfterThreshold": {},
+	"s3:ObjectRestore:Delete":                          {},
+	"s3:LifecycleTransition":                           {},
+	"s3:IntelligentTiering":                            {},
+	"s3:ObjectAcl:Put":                                 {},
+	"s3:LifecycleExpiration:*":                         {},
+	"s3:LifecycleExpiration:Delete":                    {},
+	"s3:LifecycleExpiration:DeleteMarkerCreated":       {},
+	"s3:ObjectTagging:*":                               {},
+	"s3:ObjectTagging:Put":                             {},
+	"s3:ObjectTagging:Delete":                          {},
+}
+
+func (t TopicConfiguration) Check() error {
+	if err := checkEvents(t.Events); err != nil {
+		return err
+	}
+
+	// TODO add check of ARN -- sending of test messages to SNS topic
+
+	return nil
+}
+
+func (q QueueConfiguration) Check() error {
+	if err := checkEvents(q.Events); err != nil {
+		return err
+	}
+
+	// TODO add check of ARN -- sending of test messages to SQS topic
+
+	return nil
+}
+
+func (l LambdaFunctionConfiguration) Check() error {
+	if err := checkEvents(l.Events); err != nil {
+		return err
+	}
+
+	// TODO add check of ARN
+
+	return nil
+}
+
+func checkEvents(events []string) error {
+	for _, e := range events {
+		if _, ok := validEvents[e]; !ok {
+			return errors.GetAPIError(errors.ErrEventNotification)
+		}
+	}
+	return nil
+}
+
+func (n *NotificationConfiguration) CheckAndComplete() (bool, error) {
+	completed := false
+	if len(n.TopicConfigurations) != 0 {
+		for i, t := range n.TopicConfigurations {
```

Comment:
Since we [support only](https://github.com/nspcc-dev/neofs-s3-gw/issues/340#issue-1120508149) `QueueConfiguration` we should return error.